docs/reference/mapping/types/nested.asciidoc

```diff
@@ -71,10 +70,15 @@ GET my_index/_search
 ==== Using `nested` fields for arrays of objects
 
 If you need to index arrays of objects and to maintain the independence of
-each object in the array, you should use the `nested` datatype instead of the
-<<object,`object`>> datatype.  Internally, nested objects index each object in
+each object in the array, use the `nested` datatype instead of the
+<<object,`object`>> datatype.
+
+TIP: If you consider creating `nested` objects with two `key` and `value` keyword fields, consider using the <<flattened>> datatype instead.
+Because nested documents are indexed as separate documents, they can only be accessed within the scope of the nested query. While a flattened object provides only a single field to search on, the object's contents can still be searched using simple queries and aggregations.
```

Comment:
A suggestion for how to restructure this tip:
* We could first give some context, saying that when ingesting key-value pairs with a large arbitrary set of keys, one technique is to model each pair as its own nested document with `key` and `value` fields.
* Instead we'd suggest the flattened datatype, "which maps an entire object as a single field and allows for simple searches over its contents."

One last comment -- instead of describing the downside as "only be accessed within the scope of the nested query", I think it'd be clearer to mention that nested documents and queries are generally expensive, and that the flattened datatype is a better fit for this use case.

Reply:
Thanks @jtibshirani -- I made several changes to this Tip that incorporates your feedback. See my [latest commit](https://github.com/elastic/elasticsearch/pull/56794/commits/94f92b2a2b8b0ea9249e3a072a70a3414fd23217#diff-0c297478c1696c4ffaff0240e8e3da3dR168-R169) for specific changes.